eos_data_distribution/ndn/http.py

```diff
@@ -118,6 +120,10 @@ def _got_buf(self, data, buf):
     else:
         name = re.sub('https?://', '', args.url)
 
+    loop = GLib.MainLoop()
     producer = Producer(name, args.url, auto=True)
+    if args.output:
+        consumer = file.FileConsumer(name, filename=args.output, auto=True)
+        consumer.connect('complete', lambda  *a: loop.quit())
```

Comment:
it was to have an all-inclusive test, but yes, i think this is where your modules entry points make sense.